test/benchmark/benchmark_helper.rb

```diff
@@ -0,0 +1,28 @@
+require 'benchmark/ips'
+
+require File.expand_path('../../../config/environment', __FILE__)
+require 'factory_girl_rails'
+
+unless Rails.env.production? && ActiveRecord::Base.connection.class == ActiveRecord::ConnectionAdapters::SQLite3Adapter && Rails.configuration.database_configuration[Rails.env]['database'] == ":memory:"
```

Comment:
You can't do that check and expect the error message to be printed out if it's not true, as that adapter won't be loaded.
e.g: my production.yml contains `adapter: mysql`, then I get an error `uninitialized constant ActiveRecord::ConnectionAdapters::SQLite3Adapter (NameError)` when trying to run this code instead of the message